python-backend/app/api/mines/notification/resources/notification.py

```diff
@@ -0,0 +1,51 @@
+from flask_restplus import Resource
+from sqlalchemy_filters import apply_sort
+
+from app.extensions import api
+from app.api.utils.include.user_info import User
+from ....utils.access_decorators import requires_role_mine_view
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+from ..models.notification import Notification
+from ...mine.models.mine import Mine
+
+
+class MineNotificationResource(Resource, UserMixin, ErrorMixin):
+    @api.doc(params={'mine_guid': 'Mine guid.'})
+    @requires_role_mine_view
+    def get(self):
+        user_name = User().get_user_username()
+        mine_query = Mine.query.filter_by( deleted_ind=False ).join(Notification).filter_by(idir=user_name)
+        sort_criteria = [{'model': 'Mine', 'field': 'mine_name', 'direction': 'asc'}]
+        mine_query = apply_sort(mine_query, sort_criteria)
+        mines = mine_query.all()
```

Comment:
There should be at least one safety check here. If any of these steps fail, we'll just throw a 500 (assuming something like `None.all()` results), which is undesirable.